doc/overviews/inference.md

```diff
@@ -0,0 +1,401 @@
+Users can configure which gateways, listeners, routes or paths that an inference policy should apply to.
+This is done via the policy resources (i.e. PromptGuardPolicy, TokenRateLimitPolicy) and how they target Gateways and HTTPRotues.
+Some service configuration can then be set differently per policy.
+For example, having a different set of risk categories being checked for a specifc HTTPRoute, rather than the defaults across the entire set of Gateways.
+This configuration is passed from the wasm-shim to the inference service at request time as additional context.
+Here is a request flow diagram for this integration.
+
+![ai_service_gateway_api_kuadrant](./Arch%20Overview%20v1%20-%20ai_service_gateway_api_kuadrant.jpg)
+
+Any additional context is passed to the inference service as request headers.
+For example, `x-kuadrant-guard-categories: harm,violence`.
+The inference service will detect these optional headers and change the execution path accordingly.
+One security consideration with using headers to pass the context is exploitation by users/cients.
+In particular when running where there is no wasm-shim in between the user and the inference service.
+To alleviate this risk, the set of known headers should be removed by Envoy proxy using the `request_headers_to_remove` [configuration](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route.proto).
+
+__NOTE:__ At this time, the token based rate limiting feature is only available when integrating with the Kuadrant operator. This is because of how limiting is applied by limitador at the wasm-shim layer, not the inference service layer. The inference service is responsible for gathering and reporting on token usage stats, not the actual limiting.
+
+TODO: Add details on token rate limting implemenation after these issues are further along:
+
+* https://github.com/Kuadrant/kuadrant-operator/issues/1341
+* https://github.com/Kuadrant/kuadrant-operator/issues/1342
+* https://github.com/Kuadrant/kuadrant-operator/issues/1343
+
+### Wasm-shim configuration
+
+A Kuadrant wasm-shim configuration for one PromptGuardPolicy custom resources targeting a HTTPRoute looks like the following and is generated automatically by the Kuadrant control plane.
+
+```yaml
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: kuadrant-kuadrant-ingressgateway
+  namespace: gateway-system
+spec:
+  phase: STATS
+  pluginConfig:
+    services:
+      inference-service:
+        type: ext_proc
+        endpoint: kuadrant-inference-service
+        failureMode: deny
+    actionSets:
+      - name: some_name_0
+        routeRuleConditions:
+          hostnames:
+            - "*.models.website"
+            - "*.models.io"
+          predicates:
+            - request.url_path.startsWith("/openai/v1/completions")
+        actions:
+          - service: inference-service
+            scope: gateway-system/app-promptguard
```

Comment:
@jasonmadigan @eguzki I've updated the doc and example WasmPlugin to use a single port for the inference service.
I went back on this decision so as to avoid multiple requests to the same service if more than 1 inference feature is being applied (and potentially causing multiple request body parsing).

However, I would like your thoughts on how to solve a different problem introduced by this change.
The `scope` of an action would be the originating policy. However, there may be more than 1 policy that results in an action for calling the inference-service. How to avoid multiple actions at this configuration level?